astropy/table/tests/test_table.py

```diff
@@ -3279,3 +3279,33 @@ def test_add_list_order():
     array = np.empty((20, 1))
     t.add_columns(array, names=names)
     assert t.colnames == names
+
+
+def test_table_write_preserves_nulls(tmp_path):
+    """Ensures that upon writing a table, the fill_value attribute of a
+    masked (integer) column is correctly propagated into the TNULL parameter
+    in the FITS header"""
+
+    # Could be anything except for 999999, which is the "default" fill_value
+    # for masked int arrays
+    NULL_VALUE = -1
+
+    # Create table with an integer MaskedColumn with custom fill_value
+    c1 = MaskedColumn(
+        name="a",
+        data=np.asarray([1, 2, 3], dtype=np.int32),
+        mask=[True, False, True],
+        fill_value=NULL_VALUE,
+    )
+    t = Table([c1])
+
```

Comment:
Can you add a specific unit test of the `Table.as_array()` functionality here by doing `tn = t.as_array()` and confirm the correct `fill_value`?

Reply:
Done (see aab7ab6b30d90a0788af662064154a1b9ae4a5ee)

Reply:
It seems this commit was lost in one of the rebase ?

Reply:
I rebased and added the commit.